app.go

```diff
@@ -453,28 +464,92 @@ func ValidateApp(opts ...Option) error {
 // registered via Invoke options. See the documentation of the App struct for
 // details on the application's initialization, startup, and shutdown logic.
 func New(opts ...Option) *App {
-	logger := fxlog.DefaultLogger(os.Stderr)
-
 	app := &App{
-		log:          logger,
+		log:          new(logBuffer),
```

Comment:
We don't need to initialize this field here because it's always overwritten (L481-493).
```suggestion
```